langkit/topics.py

```diff
@@ -12,16 +13,23 @@
 classifier = pipeline("zero-shot-classification", model=model_path)
 
 
-@register_metric_udf(col_type=String)
-def closest_topic(text: str) -> str:
-    output = classifier(text, _topics, multi_label=False)
-    return output["labels"][0]
+def closest_topic(text):
+    index = text.columns[0] if isinstance(text, pd.DataFrame) else list(text.keys())[0]
```

Comment:
This looks like its taking the first column from a collection? Can you explain this a little more and how it works? Might be worth a code comment as a hint for future readers of the code.

Reply:
That line shows up in several places. We're registering the same UDF for both the prompt and response columns. There's only one input column, so that grabs the unknown column name from either the data frame or dictionary.